application/Common/Shared/Events/Queue/ISupportDeserializationEventQueue.cs

```diff
@@ -6,22 +6,22 @@ namespace MORR.Shared.Events.Queue
     ///     Provides a single-writer-multiple-reader queue for <see cref="Event" /> types with support for deserialization.
     /// </summary>
     /// <typeparam name="T">The type of the event</typeparam>
-    public interface ISupportDeserializationEventQueue<out T> where T : Event
+    public interface ISupportDeserializationEventQueue<out T>: IReadOnlyEventQueue<T> where T : Event
```

Comment:
I would prefer to keep the `ISupportDeserializationEventQueue` and `IReadOnlyEventQueue` interfaces completely separate. Currently the only feasible mechanism for filtering out events during processing would be to create an event queue that supports deserialization but does not implement `IReadOnlyEventQueue` (so it would not get picked up by `JsonIntermediateFormatSerializer`) - we don't provide such an implementation, but I still think it would be disadvantageous to disallow this in the interface.

Reply:
Should be resolved now.